wdl-engine/src/backend/crankshaft.rs

```diff
@@ -397,41 +399,69 @@ impl TaskExecutionBackend for CrankshaftBackend {
         Some(Path::new(GUEST_WORK_DIR))
     }
 
-    fn localize_inputs<'a, 'b, 'c, 'd>(
+    fn localize_inputs<'a, 'b, 'c>(
         &'a self,
-        downloader: &'b dyn Downloader,
+        downloader: HttpDownloader,
         inputs: &'c mut [crate::eval::Input],
-    ) -> BoxFuture<'d, Result<()>>
+    ) -> BoxFuture<'c, Result<()>>
     where
-        'a: 'd,
-        'b: 'd,
-        'c: 'd,
-        Self: 'd,
+        'a: 'c,
+        'b: 'c,
+        Self: 'c,
     {
-        async {
+        async move {
             // Construct a trie for mapping input guest paths
             let mut trie = InputTrie::default();
             for input in inputs.iter() {
                 trie.insert(input)?;
             }
 
             for (index, guest_path) in trie.calculate_guest_paths(GUEST_INPUTS_DIR)? {
-                inputs[index].set_guest_path(guest_path);
+                if let Some(input) = inputs.get_mut(index) {
+                    input.set_guest_path(guest_path);
+                } else {
+                    bail!("invalid index {} returned from trie", index);
+                }
             }
 
             // Localize all inputs
             // TODO: only do this for local task execution
```

Comment:
I'm unsure what this comment means. @peterhuene, can this comment be deleted once this is merged in?

Reply:
I'm going to say, for now, let's keep it in.